app/controllers/users_controller.rb

```diff
@@ -2,5 +2,11 @@ class UsersController < ApplicationController
 
   def show
     @user = User.find(params[:id])
+    if session[:user_id] == @user.id
+      render 'show'
+    else
+      session[:user_id] = nil
+      redirect_to login_path
+    end
```

Comment:
Agreed. It's reminiscent of what Facebook did to Mike when he tried changing stuff in his cookies.